uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/directives/spinner.js

```diff
@@ -1,90 +1,84 @@
+/* eslint-env amd */
+
 define([
-    'tasks-assignments/directives/directives'
+  'tasks-assignments/directives/directives'
 ], function (directives) {
-    'use strict';
-
-    directives.directive('ctSpinner',['$rootScope','$log',function ($rootScope, $log) {
-        $log.debug('Directive: ctSpinner');
-
-        return {
-            link: function ($scope, el, attrs) {
-                var cover = document.createElement('div'),
-                    positionSet = false,
-                    coverSet = false,
-                    spinner;
-
-                cover.className = 'ct-spinner-cover ct-spinner-img';
-
-                function isPositioned(){
-                    var elPosition = window.getComputedStyle(el[0]).position;
-                    return elPosition == 'relative' || elPosition == 'absolute' || elPosition == 'fixed'
-                }
-
-                function appendSpinner() {
-                    if (!isPositioned()) {
-                        el.css('position','relative');
-                        positionSet = true;
-                    }
-
-                    el.append(cover);
-                    coverSet = true;
-                }
+  'use strict';
 
-                function removeSpinner(){
-                    coverSet && cover.parentNode.removeChild(cover);
-                    coverSet = false;
+  directives.directive('ctSpinner', ['$rootScope', '$log', function ($rootScope, $log) {
+    $log.debug('Directive: ctSpinner');
 
-                    if (positionSet) {
-                        el.css('position','');
-                    }
-                }
+    return {
+      link: function ($scope, el, attrs) {
+        var cover = document.createElement('div');
+        var positionSet = false;
+        var coverSet = false;
+        var spinner;
 
-                if (typeof attrs.ctSpinnerShow !== 'undefined') {
-                    appendSpinner();
-                }
+        cover.className = 'ct-spinner-cover spinner';
 
-                if (typeof attrs.ctSpinnerMainView !== 'undefined') {
-                    cover.className = 'ct-spinner-cover';
-
-                    spinner = document.createElement('div');
-                    spinner.className = 'ct-spinner ct-spinner-img';
-
-                    $rootScope.$on('$stateChangeStart', function() {
-                        appendSpinner();
-                        document.body.appendChild(spinner);
-                    });
+        function isPositioned () {
```

Comment:
No func description. Same issue as in https://github.com/civicrm/civihr/pull/2333

Reply:
Updated!